substrate/frame/staking/src/lib.rs

```diff
@@ -1223,3 +1223,67 @@ impl BenchmarkingConfig for TestBenchmarkingConfig {
 	type MaxValidators = frame_support::traits::ConstU32<100>;
 	type MaxNominators = frame_support::traits::ConstU32<100>;
 }
+
+/// Controls validator disabling
+pub trait DisablingStrategy<T: Config> {
+	/// Make a decision if an offender should be disabled or not. The result is a `Vec` of validator
+	/// indices that should be disabled
+	fn make_disabling_decision(
+		offender_stash: &T::AccountId,
+		slash_era: EraIndex,
+		currently_disabled: &Vec<u32>,
+	) -> Vec<u32>;
```

Comment:
In which case we would return a vec of length more than 1? If there are no such cases, this can be Option<u32> instead or an enum.

Reply:
With the current implementation there is no such case. The return type is `Vec` because another implementation might decide to disable more than one validator per decision.

Previously there was support for re-enabling offenders but we dropped it. I'm replacing this `Vec` with an `Option` because the current code looks weird (e.g. https://github.com/paritytech/polkadot-sdk/pull/2226/files#r1451798760).